src/rs.c

```diff
@@ -437,8 +437,10 @@ hsk_rs_respond(
   hsk_rs_log(ns, "  secure: %d\n", result->secure);
   hsk_rs_log(ns, "  bogus: %d\n", result->bogus);
 
-  if (result->why_bogus)
+  if (result->bogus) {
     hsk_rs_log(ns, "  why_bogus: %s\n", result->why_bogus);
+    goto fail;
```

Comment:
Needs a check for the CD bit in the request here. If it's set then return the response as is even if bogus. 

```
dig @hnsd-recursive dnssec-failed.org
SERVFAIL
dig @hnsd-recursive dnssec-failed.org +cd
dnssec-failed.org.      7200    IN      A       69.252.80.75
```

Reply:
addressed in 5e014871b6eb59c0540ca876993a860dffac9147 in PR #86